src/client/index.js

```diff
@@ -388,6 +389,31 @@ function mkcircuit(data, opts) {
     show_scale();
     monitorview.on('change:start', show_range);
     monitorview.on('change:pixelsPerTick', show_scale);
+
+
+    $('button[name=zoom-in]').click(e => {
+        // Zoom only if we already have a circuit on the paper
+        if (typeof paper.scale === 'function') {
+            const { sx: currentScale } = paper.scale();
+            paper.scale(1.1 * currentScale);
```

Comment:
Accumulating scale value by multiplying by a floating point number is bound to create inaccuracies, i.e. using the buttons multiple times back and forth will not return to the original scale value. Please use an integer, and calculate the floating point factor from it.

Reply:
I think the good solution would be to zoom-in by adding 0.1 to the scale: `paper.scale(currentScale + 0.1)`
and subtracting 0.1 when zooming-out.

Is it a good idea?

Reply:
Fixed in [this commit](https://github.com/tilk/digitaljs_online/pull/34/commits/7c3ebca2de5160fffce756a398be8ccb3791aec4).

Reply:
This changes the behavior of the scaling buttons - the scaling is now linear. Users generally expect that scaling buttons multiply the scaling by a constant factor - e.g. exponential scaling.

Reply:
How should I solve this? Unfortunately I didn't understand the suggestion about calculating a factor.

Reply:
Do you know what an exponential function is?

Reply:
If you click the `+` button three times, the scaling factor is 1.1 to the power of 3. In my original comment, I suggested to remember the three, and calculate the power. Is this clear?